Merge Development

bayurisma opened the pull request.

bayurisma committed:
feat: add jwt implementation and some auth related endpoint

bayurisma committed:
feat: add auth middleware

bayurisma committed:
refactor: separating some logic into different function

bayurisma committed:
fix: change status code for error 401

bayurisma committed:
Merge branch 'main' into development

hugo-setiawan commented on src/routes/index.ts:
```diff
@@ -1,10 +1,12 @@
 // Mount the routers
-router.use("/sample", sampleRouter);
+router.use("/sample", sampleRouter)
+router.use("/api/v1", authRouter)
```
maybe we want to move the auth router to `/api/v1/auth`, we could also refactor and create a main api router for the `/api/v1` path

hugo-setiawan commented on src/errors/responseError.ts:
```diff
@@ -0,0 +1,9 @@
+  statusCode: number,
+  message: string
+) => {
+  return res.status(statusCode).json({ status: "ERROR", message })
```
for the status message in the json, maybe we can use something like [http-status-codes](https://www.npmjs.com/package/http-status-codes) reason phrase to automatically set the message based on `statusCode`

hugo-setiawan reviewed (changes requested):
thanks for the work, and for the refactor as well! some comments before we can move forward

bayurisma committed:
fix and refactor :
- create main router path and add /auth for authRouter path
- use http-status-code to provide status error message based on given status code

hugo-setiawan commented:
LGTM!

hugo-setiawan committed:
feat: add prisma migrations for entire schema
